packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,43 +7,81 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	daysSelected: readonly [Date, Date];
 	disabled?: boolean;
+	range?: boolean;
```

Comment:
I think not we have compatibility issues here because `daysSelected` is an internal state of the component, so we have control over that. Well, what I thought about this is that `daysSelected` behaves like a tuple but when it is not a range, we will have only `[Date]` and when it is range `[Date, Date]`.

Currently, this code is not aligned in this sense, it only considers that it will always be a tuple with two dates, so when it is not a range, the dates will be the same.